lib/wallaroo/core/topology/router.pony

```diff
@@ -1133,23 +1405,31 @@ class val LocalPartitionRouter[In: Any val,
               Fail()
               (true, latest_ts)
             end
-          | let p: ProxyRouter =>
-            p.route[D](metric_name, pipeline_time_spent, data, producer_id,
-              producer, i_msg_uid, frac_ids, latest_ts, metrics_id,
-              worker_ingress_ts)
-          end
-        else
-          ifdef debug then
-            match key
-            | let k: Stringable val =>
-              @printf[I32](("LocalPartitionRouter.route: No entry for " +
-              "key %s\n\n").cstring(), k.string().cstring())
-            else
+          else
+            ifdef debug then
               @printf[I32](("LocalPartitionRouter.route: No entry for " +
-              "this key\n\n").cstring())
+                "key '%s'\n\n").cstring(), key.string().cstring())
             end
+            let routing_args = TypedRoutingArguments[D](metric_name,
+              pipeline_time_spent, data, producer_id, i_msg_uid,
+              frac_ids, latest_ts, metrics_id, worker_ingress_ts)
+            producer.unknown_key(_state_name, key, routing_args)
+            (true, latest_ts)
```

Comment:
This should be `(false, ...` because we're not done with this message.

Reply:
done